### PR TITLE
Using Doxygen for C++ docs

### PR DESCRIPTION
**Pull request by robertbastian**
Based on and similar to #4371 

https://icu4x-pr-artifacts.storage.googleapis.com/gha/9abdfccb4de9cff746b8234241c1b10857464831/ffi/cpp/index.html

### CODE REVIEW

**Comment by Manishearth** [comment]
Doc nits:

 - `template<typename W >` methods should include a line in their docs linking to Writeable
 - ICU4XError is not linked

**Review by Manishearth — APPROVED** [review]

**Comment by robertbastian** [comment]
> template<typename W > methods should include a line in their docs linking to Writeable

This is an issue with the CPP inline docs, not with the doxygen setup, right?

**Comment by Manishearth** [comment]
Yes

**Comment by Manishearth** [comment]
Should be fixed in Diplomat, I mean.

**Review by Manishearth — APPROVED** [review]
We should probably include a sample doxygen config in DIplomat too.

**Comment by sffc** [comment]
The docs preview link (from artifacts-info.yml) isn't working

**Comment by robertbastian** [comment]
The action didn't get run on that commit, maybe because of merge conflicts
image

File: .github/workflows/artifacts-build.yml [hunk 1]
```diff
@@ -126,12 +116,17 @@ jobs:
         node-version: 16.18.0
         cache: 'npm'
         cache-dependency-path: '**/package-lock.json'
-
-    - name: Build CPP docs
+    - name: Install doxygen-awesome
       run: |
         cd ffi/capi/cpp/docs
-        make html
+        git clone https://github.com/jothepro/doxygen-awesome-css.git
```
Comment (sffc) [inline .github/workflows/artifacts-build.yml:122]: Nit (optional)
```suggestion
        git clone --depth 1 https://github.com/jothepro/doxygen-awesome-css.git
```

**Review by sffc — APPROVED** [review]

**Pull request merged**